### PR TITLE
refactor(core): refactor openapi docs for protected app

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

refactor openapi docs for protected app, to align with other APIs.

## Testing

N/A

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/routes/applications/application-protected-app-metadata.openapi.json [hunk 1]
```diff
@@ -2,20 +2,20 @@
   "paths": {
     "/api/applications/{id}/protected-app-metadata/custom-domains": {
       "get": {
-        "summary": "Get the list of custom domains of the protected application.",
-        "description": "Get the list of custom domains of the protected application. This feature is not available for open source version.",
+        "summary": "Get application custom domains.",
+        "description": "Get custom domains of the specified application, the applicatioin type should be protected app.",
```
Comment (charIeszhao) [inline packages/core/src/routes/applications/application-protected-app-metadata.openapi.json:6]:
```suggestion
        "description": "Get custom domains of the specified application, the application type should be protected app.",
```

File: packages/core/src/routes/applications/application-protected-app-metadata.openapi.json [hunk 2]
```diff
@@ -2,20 +2,20 @@
...
         "responses": {
           "200": {
-            "description": "The domain list of the protected application."
+            "description": "An array of custom domains of the application."
```
Comment (charIeszhao) [inline packages/core/src/routes/applications/application-protected-app-metadata.openapi.json:9]:
```suggestion
            "description": "An array of the application custom domains."
```

File: packages/core/src/routes/applications/application-protected-app-metadata.openapi.json [hunk 3]
```diff
@@ -44,11 +44,11 @@
     },
     "/api/applications/{id}/protected-app-metadata/custom-domains/{domain}": {
       "delete": {
-        "summary": "Delete a custom domain.",
-        "description": "Add a custom domain. This feature is not available for open source version.",
+        "summary": "Remove a custom domain.",
+        "description": "Remove a custom domain from the specified application.",
```
Comment (charIeszhao) [inline packages/core/src/routes/applications/application-protected-app-metadata.openapi.json:48]:
```suggestion
        "summary": "Remove custom domain.",
        "description": "Remove custom domain from the specified application.",
```

File: packages/core/src/routes/applications/application-protected-app-metadata.openapi.json [hunk 4]
```diff
@@ -44,11 +44,11 @@
     },
     "/api/applications/{id}/protected-app-metadata/custom-domains/{domain}": {
       "delete": {
-        "summary": "Delete a custom domain.",
-        "description": "Add a custom domain. This feature is not available for open source version.",
```
Comment (charIeszhao) [inline packages/core/src/routes/applications/application-protected-app-metadata.openapi.json:48]: 🤣

File: packages/core/src/routes/applications/application-protected-app-metadata.openapi.json [hunk 5]
```diff
@@ -2,20 +2,20 @@
...
           "400": {
             "description": "Faild to sync the domain info from remote provider."
           }
         }
       },
       "post": {
-        "summary": "Add a custom domain to the protected application.",
-        "description": "Add a custom domain to the protected application. You'll need to setup DNS record later. This feature is not available for open source version.",
+        "summary": "Add a custom domain to application.",
+        "description": "Add a custom domain to the application. You'll need to setup DNS record later.",
```
Comment (charIeszhao) [inline packages/core/src/routes/applications/application-protected-app-metadata.openapi.json:18]:
```suggestion
        "summary": "Add custom domain to the application.",
        "description": "Add custom domain to the application. You'll need to setup DNS record later.",
```

**Review by charIeszhao — APPROVED** [review]

**Review by simeng-li — APPROVED** [review]

**Pull request merged**